engine/engine.go

```diff
@@ -228,6 +227,9 @@ loop:
 
 				for _, vector := range r {
 					for i := range vector.Samples {
+						if len(series[i].Points) == 0 {
```

Comment:
What is the reason for moving the allocation here?

Reply:
As in this case I'm adding all series as output but only filling some, i wanna avoid this memory allocation (so i only allocate when I'm sure there are samples on it).

This reduced a lot the memory allocation on the benchmark.